src/formatDate.js

```diff
@@ -8,7 +8,38 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat[fromFormat.length - 1];
+  const newSeparator = toFormat[toFormat.length - 1];
+  const dateParts = date.split(`${oldSeparator}`);
+  const dateObject = {};
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    dateObject[fromFormat[i]] = dateParts[i];
+  }
+
+  if (dateObject.YY) {
+    const year = Number(dateObject.YY);
+
+    dateObject.YYYY = year < 30 ? '20' + dateObject.YY : '19' + dateObject.YY;
+  }
```

Comment:
When converting from 'YY' to 'YYYY', the code correctly prefixes the year with '20' or '19'. However, it's important to ensure that the conversion is only done if the 'toFormat' array requires a 'YYYY' format. As it stands, the conversion will always occur if 'YY' is present in the 'fromFormat', which may not be necessary if the 'toFormat' also uses 'YY'. You should check if 'YYYY' is included in the 'toFormat' before performing this operation.